main.py

```diff
@@ -65,33 +85,67 @@ def do_create_room(self, arg):
         Dojo().create_room(arg)
 
     @docopt_cmd
-    def do_add_person(self,arg):
+    def do_add_person(self, arg):
         """Usage: add_person <person_fname> <person_lname> <FELLOW/STAFF> [<wants_accommodation>]"""
         check = Dojo().add_person(arg)
-        if check == True:
-                Dojo().assign_room(arg)
+        if check:
+            Dojo().allocate_room(arg)
         else:
-            pass
+            print("Dojo Couldn't Allocate a room")
 
     @docopt_cmd
-    def do_print_allocations(self,arg):
-        """Usage: print_allocations [<-o=filename>] """
+    def do_print_allocations(self, arg):
+        """Usage: print_allocations [--o=filename] """
         Dojo().print_allocations(arg)
 
     @docopt_cmd
-    def do_print_room(self,arg):
+    def do_print_unallocated(self, arg):
+        """Usage: print_unallocated [--o=filename] """
+        Dojo().print_unallocated(arg)
+
+    @docopt_cmd
+    def do_get_person_id(self, arg):
+        """Usage: get_person_id <person_fname> <person_lname> """
+        Dojo().get_person_id(arg)
+
+    @docopt_cmd
+    def do_print_room(self, arg):
         """Usage: print_room <room_name>"""
         Dojo().print_room(arg)
 
+    @docopt_cmd
+    def do_reallocate_person(self, arg):
+        """Usage: reallocate_person <person_identifier> <new_room_name>"""
+        Dojo().reallocate_person(arg)
+
+    @docopt_cmd
+    def do_load_people(self, arg):
+        """Usage: load_people <file_name>"""
+        Dojo().load_people(arg)
+
+    @docopt_cmd
+    def do_save_state(self, arg):
+        """Usage: save_state [--db=sqlite_database]"""
+        DatabaseManager().save_state(arg)
+
+    @docopt_cmd
+    def do_load_state(self, arg):
+        """Usage: load_state <sqlite_database>"""
+        DatabaseManager().load_state(arg)
 
     @docopt_cmd
-    def do_db_init(self,arg):
-        """Initialize the database for first time use. \
-        (This should only be done once to avoid complications."""
-        DatabaseInit().initialize()
+    def do_purge(self, arg):
+        """Usage: purge"""
+        print("Are you sure you want to purge the dojo?")
+        arg = input("Enter \'yes\' to approve and 'no' to reject: \n ")
+        Dojo().purge(arg)
```

Comment:
maybe only purge when the `arg` is a `yes` instead of checking for the yes in the "backend"